modelscope/trainers/hooks/checkpoint/checkpoint_processor.py

```diff
@@ -175,20 +178,22 @@ def link(self, model, src_file, output_dir):
                 'changing to copy the bin file, this may use more disk space.')
             shutil.copyfile(src_file, dest_file)
 
-    def save_trainer_state(self, trainer, model, train_state_file, meta):
+    def save_trainer_state(self, trainer, model, train_state_file, meta,
+                           save_optimizers):
         """Save the trainer state, including optimizer/lr_scheduler's state dict, random states etc.
 
         Args:
             trainer: The trainer instance.
             model: The model instance.
             train_state_file: The target file name for saving trainer states.
             meta: Some extra meta info.
+            save_optimizers: Save optimizers state or not.
         """
         save_checkpoint(
             model,
             train_state_file,
-            trainer.optimizer,
-            trainer.lr_scheduler,
+            trainer.optimizer if save_optimizers else None,
+            trainer.lr_scheduler if save_optimizers else None,
```

Comment:
It appears that, apart from the optimizer and learning rate (lr), other information does not need to be periodically saved.